ocaml/libs/tracing/tracing.ml

```diff
@@ -0,0 +1,81 @@
+(*
+ * Copyright (C) 2023 Cloud Software Group
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+module SpanContext = struct
+  type t = {trace_id: string; span_id: string} [@@deriving rpcty]
+end
+
+module Span = struct
+  type t = {
+      context: SpanContext.t
+    ; parent: t option
+    ; name: string
+    ; begin_time: float
+    ; end_time: float option
+    ; tags: (string * string) list
+  }
+  [@@deriving rpcty]
+
+  let generate_id n = String.init n (fun _ -> "0123456789abcdef".[Random.int 16])
```

Comment:
Might be more efficient to generate a single int64 number and then encode it in hex, but I haven't measured. For now lets leave it as is and we can add some microbenchmarks later, see what brings up and then tweak the code if needed.

Reply:
When we start using the Opentelemetry library we can use its method of generating span/trace ids: https://ocaml.org/p/opentelemetry/latest/doc/Opentelemetry/Rand_bytes/index.html

Reply:
That would be ideal to use that library.

Reply:
CP-42794

Reply:
CP-42794